docs/products/opensearch/howto/get-started-with-nodejs.rst

```diff
@@ -0,0 +1,528 @@
+
+.. code-block:: javascript
+
+    /**
+     * Specifying a slop - a distance between search words.
+     */
+    module.exports.slopSearch = (field, query, slop) => {
+        console.log(`Searching for ${query} within distance of ${slop} in the field ${field}`);
+        const body = {
+            'query': {
+                'match_phrase': {
+                    [field]: {
+                        query,
+                        slop
+                    }
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+
+Let's use this method to find recipes for pizza with pineapple. I've learned from my Italian colleagues that it is an illegal combination! Let's see if we have any recipes where words pizza and pineapple are located within the distance of maximum 10 words.
+
+::
+
+    run-func index.js slopSearch directions "pizza pineapple" 10
+
+And we've found "Pan-Fried Hawaiian Pizza" ;)
+
+Another useful feature of full-text search are fuzzy queries, which are used to take into account typos and misspellings.
+
+.. code-block:: javascript
+
+    /**
+     * Specifying fuzziness to account for typos and misspelling.
+     */
+    module.exports.fuzzySearch = (value, fuzziness) => {
+        console.log(`Search for ${value} with fuzziness ${fuzziness}`);
+        const query = {
+            'query': {
+                'fuzzy': {
+                    "title": {
+                        value,
+                        fuzziness
+                    }
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body: query
+        }, logTitles)
+    };
+
+::
+
+    run-func index.js fuzzySearch "pinapple" 2
+
+And though there is a typo in the word Pineapple, we still got relevant results.
+
+Boolean queries
+^^^^^^^^^^^^^^^
+
+The last type of queries is the boolean one, useful when we want to combine multiple queries together.
+Let's find recipes to make a quick and easy dish, with no garlic, low sodium and high protein.
+
+.. code-block:: javascript
+
+    /**
+     * Combining several queries together
+     */
+    module.exports.booleanSearch = () => {
+        console.log(`Searching for food withing Salmon
+            category without onion with low sodium and high protein`);
+        const body = {
+            'query': {
+                'bool': {
+                    "must": { "match":{"categories": "Quick & Easy"}},
+                    "must_not": {"match":{"ingredients": "garlic"}},
+                    "filter": [ {"range": {"sodium": {lt: 50}}},
+                        {"range": {"protein": {gte: 5}}}],
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+::
+
```

Comment:
Let's have an explanation of the match syntaxes and how combining them works. It's pretty cool, and people might want to learn about them as well as just see them in action